nats/aio/client.py

```diff
@@ -1437,38 +1370,38 @@ async def _process_msg(self, sid, subject, reply, data):
         self.stats['in_bytes'] += payload_size
 
         sub = self._subs.get(sid)
-        if sub is None:
+        if not sub:
             # Skip in case no subscription present.
             return
 
-        sub.received += 1
-        if sub.max_msgs > 0 and sub.received >= sub.max_msgs:
+        sub._received += 1
+        if sub._max_msgs > 0 and sub._received >= sub._max_msgs:
             # Enough messages so can throwaway subscription now.
             self._subs.pop(sid, None)
         msg = self._build_message(subject, reply, data)
 
         # Check if it is an old style request.
-        if sub.future is not None:
-            if sub.future.cancelled():
+        if sub._future:
```

Comment:
arbitrary line for discussion: pre-existing, but it seems like a potential smell that client is accessing a lot of subscriptions internals. Sub being a parameter to a lot of these functions also makes me think that some of the behavior could be moved to the Subscription class. That may help isolate future changes and hide some of the complexity of these conditionals, making this class a little cleaner.

Reply:
Agreed, I got a fair amount of this code moved